Add OPT (#17088)
huggingface/transformers: docs/source/en/model_doc/opt.mdx

<!--Copyright 2020 The HuggingFace Team. All rights reserved.

Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
the License. You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
specific language governing permissions and limitations under the License.
-->

# OPT

## Overview

The OPT model was proposed in [<INSERT PAPER NAME HERE>](<INSERT PAPER LINK HERE>) by <INSERT AUTHORS HERE>.
<INSERT SHORT SUMMARY HERE>

The abstract from the paper is the following:

*<INSERT PAPER ABSTRACT HERE>*

Tips:

<INSERT TIPS ABOUT MODEL HERE>

This model was contributed by [INSERT YOUR HF USERNAME HERE](https://huggingface.co/<INSERT YOUR HF USERNAME HERE>).
The original code can be found [here](<INSERT LINK TO GITHUB REPO HERE>).

## OPTConfig

[[autodoc]] OPTConfig
    - all

> ## OPTTokenizer

Reviewer: Should the OPTTokenizer be removed to pass the tests? On error is on serialising the .mdx file

<!--Copyright 2022 The HuggingFace Team. All rights reserved.

Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
the License. You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
specific language governing permissions and limitations under the License.
-->

# OPT

## Overview

The OPT model was proposed in [Open Pre-trained Transformer Language Models](https://arxiv.org/pdf/2205.01068) by Meta AI.

> OPT is a serie of pretrained decoder only LM which perform similarly to GPT3. 

Reviewer 2: ```suggestion
OPT is a series of open-sourced large causal language models which perform similar in performance to GPT3. 
```

The abstract from the paper is the following:

Large language models, which are often trained for hundreds of thousands of compute days, have shown remarkable capabilities for zero- and few-shot learning. Given their computational cost, these models are difficult to replicate without significant capital. For the few that are available through APIs, no access is granted to the full model weights, making them difficult to study. We present Open Pre-trained Transformers (OPT), a suite of decoder-only pre-trained transformers ranging from 125M to 175B parameters, which we aim to fully and responsibly share with interested researchers. We show that OPT-175B is comparable to GPT-3, while requiring only 1/7th the carbon footprint to develop. We are also releasing our logbook detailing the infrastructure challenges we faced, along with code for experimenting with all of the released models.

Tips:

> If you want to use the `opt-350m`, you need to set the `do_layer_norm_before` parameter to `False`. 

Reviewer 2: ```suggestion
```

This is done automatically in the config. The comment would confuse users IMO

> This model was contributed by [Patrick Von Platen](https://huggingface.co/patrickvonplaten), [Arthur Zucker](https://huggingface.co/ArthurZ) and [Younes Belkada](https://huggingface.co/ybelkada),.

Reviewer 2: ```suggestion
This model was contributed by [Arthur Zucker](https://huggingface.co/ArthurZ) and [Younes Belkada](https://huggingface.co/ybelkada), [Patrick Von Platen](https://huggingface.co/patrickvonplaten).
```

You guys wrote the code ;-)

> This model was contributed by [Arthur Zucker](https://huggingface.co/ArthurZ) and [Younes Belkada](https://huggingface.co/ybelkada), [Patrick Von Platen](https://huggingface.co/patrickvonplaten).

Reviewer 3: ```suggestion
This model was contributed by [Arthur Zucker](https://huggingface.co/ArthurZ), [Younes Belkada](https://huggingface.co/ybelkada), and [Patrick Von Platen](https://huggingface.co/patrickvonplaten).
```

The original code can be found [here](https://github.com/facebookresearch/metaseq).

## OPTConfig

[[autodoc]] OPTConfig

>     - all
> 

Reviewer 3: ```suggestion
```
all is not necessary.

## OPTModel

[[autodoc]] OPTModel
    - forward

> ## OPTPretrainedModel
> 
> [[autodoc]] OPTPretrainedModel
>     - all
> 

Reviewer 3: ```suggestion
```

We don't document this class in general, as there is no useful docstrings.

We should document `OPTModel` however.
